src/world/area_kkj/kkj_19/AF7C00.c

```diff
@@ -4,29 +4,273 @@
+            hud_element_set_render_pos(hudElemID, D_8024655C, D_80246560_kkj_19 + 28);
+            hud_element_set_render_depth(hudElemID, 0);
+            hud_element_set_flags(hudElemID, HUD_ELEMENT_FLAGS_80 | HUD_ELEMENT_FLAGS_DISABLED);
 
-INCLUDE_ASM(s32, "world/area_kkj/kkj_19/AF7C00", func_802406C4_AF8014);
+            hudElemID = D_80246568_C8C018[0];
+            hud_element_set_alpha(hudElemID, 255);
+            hud_element_clear_flags(hudElemID, HUD_ELEMENT_FLAGS_DISABLED);
+
+            hudElemID = D_80246568_C8C018[1];
+            hud_element_set_alpha(hudElemID, 255);
+            hud_element_clear_flags(hudElemID, HUD_ELEMENT_FLAGS_DISABLED);
+
+            D_80246558_kkj_19 = 1;
+            script->functionTemp[0] = 10;
+            break;
+        case 1:
+            D_8024655C += 20;
+            if (D_8024655C >= 51) {
+                D_8024655C = 50;
+            }
+            hud_element_set_render_pos(D_80246568_C8C018[0], D_8024655C, D_80246560_kkj_19);
+            hud_element_set_render_pos(D_80246568_C8C018[1], D_8024655C, D_80246560_kkj_19 + 28);
+            if (script->functionTemp[0] != 0) {
+                script->functionTemp[0]--;
+            } else {
+                D_80246558_kkj_19 = 10;
+            }
+            break;
+        case 10:
+            hud_element_set_script(D_80246568_C8C018[0], &HES_MashAButton);
+            dgb_08_npcGroup_80246528 = 0;
+            D_8024652C = 0;
+            D_80246558_kkj_19 = 11;
+            // fallthrough
+        case 11:
+            temp = D_80246538_kkj_19[D_80246550_kkj_19];
+            if (gGameStatusPtr->pressedButtons[0] & A_BUTTON) {
+                dgb_08_npcGroup_80246528++;
+            }
+            if (temp < dgb_08_npcGroup_80246528) {
+                dgb_08_npcGroup_80246528 = temp;
+            }
+            if (D_8024652C == 2) {
+                D_8024652C = 3;
+            }
+            for (i = 0; i < 10; i++) {
+                if (D_80246578[i] != 0) {
+                    break;
+                }
+            }
+            if (i >= 10) {
+                D_8024652C = 2;
+            } else if (D_8024652C != 3) {
+                D_8024652C = 2;
+            }
+            D_80246534--;
+            if (D_80246534 == 0) {
+                D_80246534 = 5;
+                D_80246558_kkj_19 = 12;
+            }
+            break;
+        case 12:
+            if (D_80246534 != 0) {
+                D_80246534--;
+                break;
+            }
+            script->varTable[0] = dgb_08_npcGroup_80246528;
+            hud_element_free(D_80246568_C8C018[0]);
+            hud_element_free(D_80246568_C8C018[1]);
+            free_generic_entity(D_80246554);
+            return ApiStatus_DONE2;
+    }
+
+    D_80246578[D_802465A0++] = gGameStatusPtr->pressedButtons[0] & A_BUTTON;
+
+    if (D_802465A0 >= 10) {
+        D_802465A0 = 0;
+    }
+
+    evt_set_variable(script, -0x08F0D17D, D_8024652C);
```

Comment:
this hex number is actually -149,999,997. If you go to evt.h, this can be represented as EVT_AREA_VAR(3)

Reply:
Missed this one.

Reply:
A comment in evt.h says it's deprecated:
> // deprecated, use those in script_api/macros.h instead

I'll update it to `LSW(3)` as suggested in the comment, but let me know if I shouldn't.

Reply:
whoops! good catch. We plan to rename these to human-readable names later on. but yeah, what you did is good